actions/class.Runner.php

```diff
@@ -662,7 +688,18 @@ public function timeout()
         $start = $this->hasRequestParameter('start');
 
         try {
-            $serviceContext = $this->getServiceContext();
+            $serviceContext = $this->getServiceContext(false, true);
+
+            if (!$this->runnerService->isTerminated($serviceContext)) {
+                $this->endItemTimer($serviceContext);
+                $this->saveItemState($serviceContext);
+            }
+
+            $this->runnerService->check($serviceContext);
+            $this->serviceContext->init();
+
+            $this->saveItemResponses($serviceContext);
```

Comment:
`saveItemResponses` is called in different manner, but I'll factorize the context init/check